lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -111,12 +114,14 @@ def scrub(transcript)
 
       def add_extra_data(post, payment, options)
         post[:shopperEmail] = options[:shopper_email] if options[:shopper_email]
-        post[:shopperIP] = options[:shopper_ip] if options[:shopper_ip]
-        post[:shopperReference] = options[:shopper_reference] if options[:shopper_reference]
+        post[:shopperIP] = options[:ip] if options[:ip]
```

Comment:
I guess in this case, we care if `shopperIP` is nil in the payload?

Reply:
if it is nill it should not add it into the call. What is the best way to define it in ruby code ?